signing/legacy/signing.go

```diff
@@ -0,0 +1,333 @@
+// Package signing provides functions for creating and validating essential cryptographic
+// components in IOTA, such as subseeds, keys, digests and signatures.
+package signing
+
+import (
+	"errors"
+	"math"
+
+	. "github.com/iotaledger/iota.go/consts"
+	"github.com/iotaledger/iota.go/curl"
+	"github.com/iotaledger/iota.go/kerl"
+	. "github.com/iotaledger/iota.go/trinary"
+)
+
+type SpongeFunctionCreator func() SpongeFunction
```

Comment:
There seems a lot of copy and paste from the original `signing.go` file. Please consider moving the same code into a separate package or use the existing ones.